library/core/src/main/java/com/google/android/exoplayer2/extractor/flv/FlvExtractor.java

```diff
@@ -110,6 +110,7 @@ public boolean sniff(ExtractorInput input) throws IOException, InterruptedExcept
     input.peekFully(scratch.data, 0, 4);
     scratch.setPosition(0);
     int dataOffset = scratch.readInt();
+    if (dataOffset > 9) dataOffset = 9;
```

Comment:
What kind of values are you actually seeing (e.g. is it always the same huge number)? Where are these dodgy media files actually coming from?